compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -473,6 +474,36 @@ static LogicalResult setRootConfigForConvDecomposePipeline(
       IREE::Codegen::DispatchLoweringPassPipeline::Custom);
 }
 
+static LogicalResult setRootConfigForDepthwiseConvolutionDecomposePipeline(
+    mlir::FunctionOpInterface entryPointFn, linalg::LinalgOp linalgOp,
+    AIEConfig cfg) {
+  FailureOr<std::array<uint32_t, 3>> maybeInstructionSize =
+      getMatmulInstructionSize(linalgOp);
+  int64_t OW = 4;
+  int64_t OC = 4;
+  if (succeeded(maybeInstructionSize)) {
+    auto instructionSize = maybeInstructionSize.value();
+    OW = instructionSize[0];
+    OC = instructionSize[1];
+  }
+  SmallVector<int64_t> tileSizeLevel0;
+  SmallVector<int64_t> tileSizeLevel1;
+  SmallVector<int64_t> tileSizeLevel2;
+  // Note: some of the tiling dimensions are hardcoded for now.
+  if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(linalgOp)) {
+    tileSizeLevel0 = {0, OW, OC, 16, 0, 0};
+    tileSizeLevel1 = {1, OW, OC, 4, 0, 0};
+    tileSizeLevel2 = {0, 1, 0, 0, 0, 0};
```

Comment:
These tile sizes look weird to me. It doesn't have any tiling for the last two dimensions? And we usually use for loops on reduction loops. It's better to prototype with the transform dialect first, so we make sure the IR after tiling and bufferization look correct.

Reply:
Yeah, this was tiled with 1 at some point in my experimenting, but I was hitting errors in mlir-air. I agree though -- to vectorize we need to tile to 3x3 kernel dimensions. I've asked Erwei to investigate, will make this either {1,1} or {1,0}.

Reply:
Okay, I'll also take a look at the tiling strategy before @erwei-xilinx getting involved. We are currently working on changing the tiling strategy for Conv2d ops, so could borrow some to depthwise ops when it's in a good shape.